api-extractor/src/ResolvedApiItem.ts

```diff
@@ -0,0 +1,107 @@
+import ApiItem, { ApiItemKind } from './definitions/ApiItem';
+import ApiItemContainer from './definitions/ApiItemContainer';
+import { ApiTag } from './definitions/ApiDocumentation';
+import { IDocElement, IParam } from './IDocElement';
+import { IDocItem } from './IDocItem';
+import ApiJsonFile from './generators/ApiJsonFile';
+
+/**
+ * A class to abstract away the difference between an item from our public API that could be 
+ * represented by either an ApiItem or an IDocItem that is retrieved from a JSON file.
+ */
+export default class ResolvedApiItem {
+  public kind: ApiItemKind;
+  public summary: IDocElement[];
+  public remarks: IDocElement[];
+  public deprecatedMessage: IDocElement[];
+  public isBeta: boolean;
+  public params: {[name: string]: IParam};
+  public returnsMessage: IDocElement[];
+  public members: { [name: string]: ApiItem | IDocItem};
```

Comment:
This is not right. If ResolvedApiItem is the abstraction between ApiItem and IDocItem, and it has members, then its members must also be an abstraction somehow.

But I thought our idea was that ResolvedApiItem is mostly used to by inherit doc to copy the JSDoc for a single item (e.g. a class summary).  Why do we need to know the members?

The lookup mechanism needs to traverse the members, but you already have separate lookup algorithms for local vs JSON sources.  If you implement two different lookup functions, then maybe you only need to convert to ResolvedApiItem  at the last step.